cloudinit/net/networkd.py

```diff
@@ -103,19 +105,24 @@ def generate_link_section(self, iface, cfg):
 
     def parse_routes(self, conf, cfg):
         sec = 'Route'
+        route_cfg_map = {
+            'gateway': 'Gateway',
+            'network': 'Destination',
+            'metric': 'Metric',
+        }
+
         for k, v in conf.items():
-            if k == 'gateway':
-                cfg.update_section(sec, 'Gateway', v)
-            elif k == 'network':
-                tmp = v
-                if 'prefix' in conf:
-                    tmp += '/' + str(conf['prefix'])
-                cfg.update_section(sec, 'Destination', tmp)
-            elif k == 'metric':
-                cfg.update_section(sec, 'Metric', v)
+            if k == 'netmask' or k == 'prefix':
+                continue
+            if k == 'network' and 'prefix' in conf:
+                v += '/' + str(conf['prefix'])
+            elif k == 'network' and 'netmask' in conf:
+                v += '/' + network_state.mask_to_net_prefix(conf['netmask'])
```

Comment:
I don't think this will work. `mask_to_net_prefix` should return an int, so adding a string to an int like this should result int a TypeError. Did you miss an `str` call?

Can you also add/update a unit test that will take this path?

Reply:
Yes, this is not looking good. I will rework on this, thanks.

Reply:
Cleaned this up and added a test for handling duplicates